src/Pos/DB/Block.hs

```diff
@@ -221,14 +224,21 @@ putBi k v = rocksPutBi k v =<< getBlockIndexDB
 delete :: (MonadDB m) => ByteString -> m ()
 delete k = rocksDelete k =<< getBlockIndexDB
 
-getData ::  (MonadIO m, Bi v) => FilePath -> m (Maybe v)
-getData = undefined
+getData ::  (MonadIO m, MonadCatch m, Bi v) => FilePath -> m (Maybe v)
+getData fp = liftIO (decodeMaybe . BSL.fromStrict <$> BS.readFile fp) `catch` handle
+  where
+    decodeMaybe x = case decodeFull x of
+        Left _  -> Nothing
+        Right r -> Just r
+    handle (_::SomeException) = pure Nothing
```

Comment:
http://softwareengineering.stackexchange.com/a/189225

tl;dr: check whether file exists and remove it only if it does.